src/App.tsx

```diff
@@ -1,41 +1,92 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import { Post, User } from './react-app-env';
+
 import './App.scss';
 import './styles/general.scss';
+
+import { getUserPosts, getPostDetails } from './api/posts';
+import { getUsers } from './api/api';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const [currentUserId, setCurrentUserId] = useState('0');
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  const currentPost = (postId: number) => {
+    setSelectedPostId(postId);
+  };
+
+  useEffect(() => {
+    getUserPosts(currentUserId)
+      .then(response => setPosts(response));
+  }, []);
+
+  useEffect(() => {
+    getUsers()
+      .then(response => setUsers(response));
+  }, []);
+
+  useEffect(() => {
+    getUserPosts(currentUserId)
+      .then(response => setPosts(response));
+  }, [currentUserId]);
+
+  useEffect(() => {
```

Comment:
it's better to load postsdetais in apropriate component

Reply:
fix @alexandra-protyanova comment